tests/ansible/ocsp/tasks/certificate_self_validation_with_crl.yml

```diff
@@ -0,0 +1,575 @@
+    command: pki -n ocspadmin ocsp-user-show ocspadmin
+  register: ocsp_command
+  failed_when: "'User ID: ocspadmin' not in ocsp_command.stdout"
+
+- name: Identify the admin certificate serial
+  community.docker.docker_container_exec:
+    container: "{{ ca_container }}"
+    command: "pki -n caadmin  ca-cert-find --email ocspadmin@example.com"
+  register: ca_command
+
+- name: Put the OCSP admin on hold
+  community.docker.docker_container_exec:
+    container: "{{ ca_container }}"
+    command: "pki -n caadmin  ca-cert-hold {{ ca_command.stdout | regex_search('\\s*Serial Number:\\s*(\\S*)', '\\1') | first }}  --force"
+
+- name: Wait for CRL propagation
+  ansible.builtin.pause:
+    seconds: 15
+
+- name: Interact with revoked certificate
+  community.docker.docker_container_exec:
+    container: "{{ ocsp_container }}"
+    command: pki -n ocspadmin ocsp-user-show ocspadmin
+  register: ocsp_command
+  failed_when: "'PKIException: Unauthorized' not in ocsp_command.stderr"
+
+- name: Release the OCSP admin certificate
+  community.docker.docker_container_exec:
+    container: "{{ ca_container }}"
+    command: "pki -n caadmin  ca-cert-release-hold {{ ca_command.stdout | regex_search('\\s*Serial Number:\\s*(\\S*)', '\\1') | first }}  --force"
+
+- name: Wait for CRL propagation
+  ansible.builtin.pause:
+    seconds: 15
+
+- name: Interact with good certificate again
+  community.docker.docker_container_exec:
+    container: "{{ ocsp_container }}"
+    command: pki -n ocspadmin ocsp-user-show ocspadmin
+  register: ocsp_command
+  failed_when: "'User ID: ocspadmin' not in ocsp_command.stdout"
+
+- name: Identify the OCSP DS certificate serial
+  community.docker.docker_container_exec:
+    container: "{{ ca_container }}"
+    command: "pki -n caadmin  ca-cert-find --name {{ ocspds_hostname }}"
+  register: ca_command
+
+- name: Put the OCSP DS certificate on hold
+  community.docker.docker_container_exec:
+    container: "{{ ca_container }}"
+    command: "pki -n caadmin  ca-cert-hold {{ ca_command.stdout | regex_search('\\s*Serial Number:\\s*(\\S*)', '\\1') | first }}  --force"
+
+- name: Restart OCSP
+  community.docker.docker_container_exec:
+    container: "{{ ocsp_container }}"
+    command: pki-server restart --wait
+
+- name: Interact with good certificate again
+  community.docker.docker_container_exec:
+    container: "{{ ocsp_container }}"
+    command: pki -n ocspadmin ocsp-user-show ocspadmin
+  register: ocsp_command
+  failed_when: "'PKIException: Not Found' not in ocsp_command.stderr"
+
+- name: Release the OCSP DS certificate
+  community.docker.docker_container_exec:
+    container: "{{ ca_container }}"
+    command: "pki -n caadmin  ca-cert-release-hold {{ ca_command.stdout | regex_search('\\s*Serial Number:\\s*(\\S*)', '\\1') | first }}  --force"
+
+- name: Restart OCSP 2 
+  community.docker.docker_container_exec:
+    container: "{{ ocsp_container }}"
+    command: pki-server restart --wait
```

Comment:
IIRC the server has the ability to auto recover from broken DS connection, so it might not be necessary to restart the server, but it's fine to leave this in.